src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp

```diff
@@ -65,33 +65,44 @@ static void InitializeTimeSeriesNumbers_And_ThermalClusterProductionCost(
 
         // Load
         {
-            const Data::DataSeriesLoad& data = *area.load.series;
+            Data::DataSeriesLoad& data = *area.load.series;
             assert(year < data.timeseriesNumbers.height);
-            ptchro.Consommation
-              = (data.series.width != 1) ? (long)data.timeseriesNumbers[0][year] : 0; // zero-based
+            if (data.series.width == 1)
```

Comment:
As **timeseriesNumbers** was set previously in **timeseries-numbers.cpp**, it should already be set to **0** if there is only one available time series for some given area and year.
Said in another way, there should be a need for setting **timeseriesNumbers** here.

Reply:
Actually, that dependency already exists in **timeseries-numbers.cpp**. Therefore, the TS numbers provided by `TimeSeriesNumbers::Generate` are wrong when `width==1`. I agree : the fix should be applied there, not here.